tests/bettertransformer/test_bettertransformer_audio.py

```diff
@@ -141,3 +143,109 @@ def test_raise_train(self):
+        with `keep_original_model=True` for this model.
+        """
+        for model in self.all_models_to_test:
+            with tempfile.TemporaryDirectory() as tmpdirname:
+                hf_model = AutoModel.from_pretrained(model).eval()
+                bt_model = BetterTransformer.transform(hf_model, keep_original_model=keep_original_model)
+
+                bt_model = BetterTransformer.reverse(bt_model)
+                # check if no parameter is on the `meta` device
+
+                for name, param in bt_model.named_parameters():
+                    self.assertFalse(param.device.type == "meta", f"Parameter {name} is on the meta device.")
+
+                bt_model.save_pretrained(tmpdirname)
+
+                bt_model_from_load = AutoModel.from_pretrained(tmpdirname)
+
+                # check if the state dict is the same
+                # first check if the keys are the same
+                self.assertEqual(
+                    set(bt_model.state_dict().keys()),
+                    set(bt_model_from_load.state_dict().keys()),
+                )
+                # check also with HF model
+                self.assertEqual(
+                    set(hf_model.state_dict().keys()),
+                    set(bt_model_from_load.state_dict().keys()),
+                )
+
+                for key in bt_model.state_dict().keys():
+                    self.assertTrue(
+                        torch.allclose(
+                            bt_model.state_dict()[key],
+                            bt_model_from_load.state_dict()[key],
+                        )
+                    )
+
+                    self.assertTrue(
+                        torch.allclose(
+                            hf_model.state_dict()[key],
+                            bt_model_from_load.state_dict()[key],
+                        )
+                    )
+
+    @parameterized.expand([(False,)])
+    def test_invert_model_logits(self, keep_original_model=False):
+        r"""
+        Test that the inverse converted model and hf model have the same logits.
+        Since `Wav2vec2` does not support `deepcopy` we cannot test the transformation
+        with `keep_original_model=True` for this model.
+        """
+        for model in self.all_models_to_test:
+            # get hf and bt model
+            hf_model = AutoModel.from_pretrained(model)
+            # get bt model and invert it
+            bt_model = BetterTransformer.transform(hf_model, keep_original_model=keep_original_model)
+            bt_model = BetterTransformer.reverse(bt_model)
+
+            # get inputs
+            inputs = self.prepare_inputs_for_class(model)
+
+            # get outputs
+            torch.manual_seed(42)
+            output_bt = bt_model(**inputs)
+
+            torch.manual_seed(42)
+            output_hf = hf_model(**inputs)
+
+            # Assert that the outputs are the same
+            self.assertTrue(torch.allclose(output_bt[0], output_hf[0], atol=1e-3))
```

Comment:
since `keep_original_model = False`, it sounds like this test will always pass to True

Reply:
Good point, I made some modification on the base test, I propose to compare the source code and the signature of the functions `__init__`  and `forward` since it is basically the only components we touch

Reply:
Fixed in [34363bf](https://github.com/huggingface/optimum/pull/676/commits/34363bf8f9ad9c29ea0cdfae4580062afebf907d)